frontend/src/middleware.ts

```diff
@@ -3,13 +3,33 @@ import { PORT } from 'state/utils/api';
 import { getSubdomainFromURL } from 'utilities/getSubdomain';
 import type { NextRequest } from 'next/server';
 
+const PUBLIC_FILE = /\.(.*)$/;
+
 export function middleware(request: NextRequest) {
   // Only when the subdomain is set and the path is /, redirect to the map
   if (!!getSubdomainFromURL(PORT) && request.nextUrl.pathname === '/') {
     return NextResponse.redirect(new URL('/map', request.url));
   }
   // Only when the subdomain is set do not allow access
   if (!!getSubdomainFromURL(PORT)) return NextResponse.redirect(new URL('/404', request.url));
+
+  // Avoid adding locale prefixes to api or public files
+  if (
+    request.nextUrl.pathname.startsWith('/_next') ||
+    request.nextUrl.pathname.includes('/api/') ||
+    PUBLIC_FILE.test(request.nextUrl.pathname)
+  ) {
+    return;
+  }
+
+  // Set the default locale (en) as a prefix if we don't have a locale in the url
+  if (request.nextUrl.locale === 'default') {
+    const locale = request.cookies.get('NEXT_LOCALE') || 'en';
+
+    return NextResponse.redirect(
+      new URL(`/${locale}${request.nextUrl.pathname}${request.nextUrl.search}`, request.url),
+    );
+  }
```

Comment:
If we don't have the `default` locale, I don't think we need any of this. Next.js automatically redirect the user based on the detected locale: https://nextjs.org/docs/advanced-features/i18n-routing#automatic-locale-detection.

You might have some challenges with sending the locale to the API depending on the origin of the request (SSR or client). Here you can see how it was dealt with in HeCo Invest:

- https://github.com/Vizzuality/heco-invest/blob/develop/frontend/pages/_app.tsx#L60-L84
- https://github.com/Vizzuality/heco-invest/blob/develop/frontend/services/api.ts#L26-L44

Anyway, I don't think you need to work on this issue in this PR.